src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/CachedExpressionCompiler.cs

```diff
@@ -38,117 +38,109 @@ private static class Compiler<TModel, TResult>
+
+                    // model => model.Member
+                    case MemberExpression memberExpression when memberExpression.Expression == expression.Parameters[0]:
+                        return CompileFromMemberAccess(expression, memberExpression);
 
-                    var constantValue = (TResult)constantExpression.Value;
-                    return _ => constantValue;
+                    default:
+                        return CompileSlow(expression);
                 }
+            }
 
-                return null;
+            private static Func<TModel, TResult> CompileFromConstLookup(
+                ConstantExpression constantExpression)
+            {
+                // model => {const}
+                var constantValue = (TResult)constantExpression.Value;
+                return _ => constantValue;
             }
 
             private static Func<TModel, TResult> CompileFromIdentityFunc(
                 Expression<Func<TModel, TResult>> expression)
             {
-                if (expression == null)
-                {
-                    throw new ArgumentNullException(nameof(expression));
-                }
+                // model => model
 
-                if (expression.Body == expression.Parameters[0])
+                // Don't need to lock, as all identity funcs are identical.
+                if (_identityFunc == null)
                 {
-                    // model => model
-
-                    // Don't need to lock, as all identity funcs are identical.
-                    if (_identityFunc == null)
-                    {
-                        _identityFunc = expression.Compile();
-                    }
-
-                    return _identityFunc;
+                    _identityFunc = expression.Compile();
                 }
 
-                return null;
+                return _identityFunc;
             }
 
             private static Func<TModel, TResult> CompileFromMemberAccess(
-                Expression<Func<TModel, TResult>> expression)
+                Expression<Func<TModel, TResult>> expression,
+                MemberExpression memberExpression)
             {
-                if (expression == null)
+                // model => model.Member
+                if (_simpleMemberAccessCache.TryGetValue(memberExpression.Member, out var result))
                 {
-                    throw new ArgumentNullException(nameof(expression));
+                    return result;
                 }
 
-                // Performance tests show that on the x64 platform, special-casing static member and
-                // captured local variable accesses is faster than letting the fingerprinting system
-                // handle them. On the x86 platform, the fingerprinting system is faster, but only
-                // by around one microsecond, so it's not worth it to complicate the logic here with
-                // an architecture check.
```

Comment:
Did this comment block get lost? Or, did you remove it on purpose e.g. because we don't have a fingerprinting system anymore?

Reply:
All of those, plus I doubt we actually tested this with AspNetCore across bitness.

Reply:
:ok: